lib/SparkPost/SparkPost.php

```diff
@@ -1,60 +1,137 @@
 <?php
 namespace SparkPost;
+use Ivory\HttpAdapter\Configuration;
+use Ivory\HttpAdapter\HttpAdapterInterface;
 
 class SparkPost {
-	
-	private static $config;
-	private static $defaults = array(
-			'host'=>'api.sparkpost.com',
-			'protocol'=>'https',
-			'port'=>443,
-			'strictSSL'=>true,
-			'key'=>'',
-			'version'=>'v1'	
-		);
-	
-	/**
-	 * Enforce that this object can't be instansiated
-	 */
-	private function __construct(){}
-	
-	/**
-	 * Allows the user to pass in values to override the defaults and set their API key
-	 * @param Array $configMap - Hashmap that contains config values for the SDK to connect to SparkPost
-	 * @throws \Exception
-	 */
-	public static function setConfig(array $configMap) {
-		//check for API key because its required	
-		if (isset($configMap['key'])){
-			$key = trim($configMap['key']);
-			if(empty($key)){
-				throw new \Exception('You must provide an API key');
-			}
-		} else {
-			throw new \Exception('You must provide an API key');
-		}
-		self::$config = self::$defaults;
-		foreach ($configMap as $configOption => $configValue) {
-			if(key_exists($configOption, self::$config)) {
-				self::$config[$configOption] = $configValue;
-			}
-		}
-	}
-	
-	/**
-	 * Retrieves the configuration that was previously setup by the user
-	 * @throws \Exception
-	 */
-	public static function getConfig() {
-		if (self::$config === null) {	
-			throw new \Exception('No configuration has been provided');
-		}
-		return self::$config;
-	}
-	
-	public static function unsetConfig() {
-		self::$config = NULL;
-	}
+
+  public $transmission;
+
+  /**
+   * @dec connection config for making requests.
+   */
+  private $config;
+
+  /**
+   * @desc Ivory\HttpAdapter\HttpAdapterInterface to make requests through.
+   */
+  public $httpAdapter;
+
+  /**
+   * @desc Default config values. Passed in values will override these.
+   */
+  private static $apiDefaults = [
+		'host'=>'api.sparkpost.com',
+		'protocol'=>'https',
+		'port'=>443,
+		'strictSSL'=>true,
+		'key'=>'',
+		'version'=>'v1'
+	];
+
+  /**
+   * @desc sets up httpAdapter and config
+   *
+   * Sets up instances of sub libraries.
+   *
+   * @param Ivory\HttpAdapter $httpAdapter - An adapter for making http requests
+   * @param Array $settingsConfig - Hashmap that contains config values for the SDK to connect to SparkPost
+   */
+	public function __construct($httpAdapter, $settingsConfig) {
+    //config needs to be setup before adapter because of default adapter settings
+    $this->setConfig($settingsConfig);
+    $this->setHttpAdapter($httpAdapter);
+
+    $this->transmission = new Transmission($this);
+  }
+
+
+
+  /**
+   * Creates an unwrapped api interface for endpoints that aren't yet supported.
+   * The new resource is attached to this object as well as returned
+   * @return SparkPost\APIResource - the unwrapped resource
+   */
+  public function setupUnwrapped (string $endpoint) {
```

Comment:
@jasonrhodes: For what its worth, I don't really like the name of this function.  Especially since its an externally facing function, I think we should change it but I don't really know what to call it.